include/RAJA/index/RangeSegment.hpp

```diff
@@ -145,18 +156,18 @@ struct TypedRangeSegment {
   /*!
    * \return the range (end - begin) of this Segment
    */
-  RAJA_HOST_DEVICE RAJA_INLINE StorageT size() const { return m_end - m_begin; }
+  RAJA_HOST_DEVICE RAJA_INLINE DiffT size() const { return m_end - m_begin; }
 
   //! Create a slice of this instance as a new instance
   /*!
    * \return A new instance spanning *begin() + begin to *begin() + begin +
    * length
    */
-  RAJA_HOST_DEVICE RAJA_INLINE TypedRangeSegment slice(DiffT begin,
-                                                       DiffT length) const
+  RAJA_HOST_DEVICE RAJA_INLINE TypedRangeSegment slice(StorageT begin,
```

Comment:
Same question as above - should these be the stripped type?

Reply:
Thesse don't need to be stripped type as that was already handled on the return statement on 172 `return TypedRangeSegment{stripIndexType(start), stripIndexType(end)};`

Reply:
I think length should be a `DiffT`, because it's a difference, not an element.